rotkehlchen/globaldb/updates.py

```diff
@@ -477,25 +676,43 @@ def _perform_update(
                 version += 1
                 continue
 
-            try:
-                url = f'https://raw.githubusercontent.com/rotki/assets/{self.branch}/updates/{version}/updates.sql'  # noqa: E501
-                response = requests.get(url=url, timeout=DEFAULT_TIMEOUT_TUPLE)
-            except requests.exceptions.RequestException as e:
-                connection.rollback()
-                raise RemoteError(f'Failed to query Github for {url} during assets update: {str(e)}') from e  # noqa: E501
-
-            if response.status_code != 200:
-                connection.rollback()
-                raise RemoteError(
-                    f'Github query for {url} failed with status code '
-                    f'{response.status_code} and text: {response.text}',
-                )
+            assets_url = ASSETS_UPDATES_URL.format(branch=self.branch, version=version)
+            asset_collections_url = ASSET_COLLECTIONS_UPDATES_URL.format(branch=self.branch, version=version)  # noqa: E501
+            asset_collections_mappings_url = ASSET_COLLECTIONS_MAPPINGS_UPDATES_URL.format(branch=self.branch, version=version)  # noqa: E501
 
+            assets_file = _query_file_or_rollback(
```

Comment:
I am a bit confused with this rollback mechanism. Don't we use contexts here? How does this work? Rollback to what? The way the DB is accessed in this area, after the changes that happened seems dangerous

Reply:
For the updates we create an intermediate database and then copy the new assets from it. https://github.com/rotki/rotki/blob/25a3d39e9919917bdffcae36f84f70927fa17697/rotkehlchen/globaldb/updates.py#L405

We modify the database here https://github.com/rotki/rotki/blob/25a3d39e9919917bdffcae36f84f70927fa17697/rotkehlchen/globaldb/updates.py#L466 and sqlite creates a transaction that is not commited. The rollbacks undoes that queries. Also for the rollback at the end of the file, since what we do is `INSERT` and `UPDATE`s the transaction is not commited so it can be rolled back